### PR TITLE
Combine m1 and intel mac installers to a single workflow

### PR DESCRIPTION
**Pull request by cmmarslender**
Combines m1 and intel mac installers to a single matrix workflow, since we have native m1 runners now and the jobs are essentially identical. Testing on both arch mac, and the installers worked and started syncing.

### CODE REVIEW

File: build_scripts/npm_macos/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "appdmg": "^0.6.4",
```
Comment (paninaro) [inline build_scripts/npm_macos/package.json:13]: Is this being used?

Comment (cmmarslender) [reply build_scripts/npm_macos/package.json:13]: Not entirely sure, but it was in one of the previous package.json files for mac, and was in both lock files.

Comment (paninaro) [reply build_scripts/npm_macos/package.json:13]: Ahh ok, it's a dependency of electron-installer-dmg

**Review by paninaro — APPROVED** [review]
lgtm

File: build_scripts/npm_macos/package.json (hunk 1 above)
Comment (cmmarslender) [reply build_scripts/npm_macos/package.json:13]: Looks to be used by electron-installer-dmg, so it probably doesn't need to be specifically called out like it is, but used either way

```
├─┬ electron-installer-dmg@3.0.0
│ ├── appdmg@0.6.4 deduped
```

**Comment by hoffmang9** [comment]
For old times sake - lgtm

**Pull request merged**